### PR TITLE
LongformerForSequenceClassification

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adds `LongformerForSequenceClassification`

@patrickvonplaten @ibeltagy 
All the changes here are as per we discussed  in `LongformerForQuestionAnswering`.
`forward` method automatically sets global attention on cls token.

### CODE REVIEW

**Comment by ibeltagy** [comment]
This is great. Thanks, @patil-suraj

File: src/transformers/modeling_longformer.py [hunk 1]
```diff
@@ -709,6 +709,121 @@ def forward(
...
+        attention_mask[:, 0] = 2
+
+        outputs = self.longformer(
+            input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+            inputs_embeds=inputs_embeds,
+        )
+        sequence_output = outputs[0]
```
Comment (leslyarun) [inline src/transformers/modeling_longformer.py:791]: Is it better to use pooled output for sequence classification like in BertForSequenceClassification? @ibeltagy

Comment (leslyarun) [reply src/transformers/modeling_longformer.py:791]:
```
pooled_output = outputs[1]
pooled_output = self.dropout(pooled_output)
logits = self.classifier(pooled_output)
```
Something like this? @ibeltagy @patil-suraj

**Comment by leslyarun** [comment]
Is it better to use pooled output for sequence classification like in BertForSequenceClassification? @ibeltagy @patil-suraj 

```
pooled_output = outputs[1]
pooled_output = self.dropout(pooled_output)
logits = self.classifier(pooled_output)
```

**Comment by patil-suraj** [comment]
@leslyarun 
`LongformerClassificationHead` does the pooling

**Comment by leslyarun** [comment]
> @leslyarun
> `LongformerClassificationHead` does the pooling

That's great. Fine then 👍

File: src/transformers/modeling_longformer.py [hunk 2]
```diff
@@ -710,6 +710,121 @@ def forward(
...
+    """Head for sentence-level classification tasks."""
+
+    def __init__(self, config):
+        super().__init__()
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+        self.out_proj = nn.Linear(config.hidden_size, config.num_labels)
+
+    def forward(self, features, **kwargs):
+        x = features[:, 0, :]  # take <s> token (equiv. to [CLS])
```
Comment (patrickvonplaten) [inline src/transformers/modeling_longformer.py:819]: Could you replace the `x` by `hidden_states` better for lookup later and more explicit naming :-) I know we have `x` in some `modeling_....py` files but we don't want to continue this practice.

File: tests/test_modeling_longformer.py [hunk 3]
```diff
@@ -194,6 +195,23 @@ def create_and_check_longformer_for_question_answering(
         self.parent.assertListEqual(list(result["end_logits"].size()), [self.batch_size, self.seq_length])
         self.check_loss_output(result)
 
+    def create_and_check_longformer_for_sequence_classification(
```
Comment (patrickvonplaten) [inline tests/test_modeling_longformer.py:198]: Awesome that for including the tests!

File: src/transformers/modeling_longformer.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/transformers/modeling_longformer.py:819]: After this nit it should be good to merge :-)

Comment (patil-suraj) [reply src/transformers/modeling_longformer.py:819]: Sure, `hidden_states` is more clear than `x`

File: tests/test_modeling_longformer.py (hunk 3 above)
Comment (patil-suraj) [reply tests/test_modeling_longformer.py:198]: learned this from our previous discussion :)

File: src/transformers/modeling_longformer.py [hunk 4]
```diff
@@ -710,6 +710,121 @@ def forward(
...
+    """Head for sentence-level classification tasks."""
+
+    def __init__(self, config):
+        super().__init__()
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+        self.out_proj = nn.Linear(config.hidden_size, config.num_labels)
+
+    def forward(self, features, **kwargs):
+        hidden_states = features[:, 0, :]  # take <s> token (equiv. to [CLS])
```
Comment (patrickvonplaten) [inline src/transformers/modeling_longformer.py:819]:
```suggestion
        hidden_states = hidden_states[:, 0, :]  # take <s> token (equiv. to [CLS])
```

File: src/transformers/modeling_longformer.py [hunk 5]
```diff
@@ -710,6 +710,121 @@ def forward(
...
+class LongformerClassificationHead(nn.Module):
+    """Head for sentence-level classification tasks."""
+
+    def __init__(self, config):
+        super().__init__()
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+        self.out_proj = nn.Linear(config.hidden_size, config.num_labels)
+
+    def forward(self, features, **kwargs):
```
Comment (patrickvonplaten) [inline src/transformers/modeling_longformer.py:818]:
```suggestion
    def forward(self, hidden_states, **kwargs):
```

Comment (patrickvonplaten) [reply src/transformers/modeling_longformer.py:818]: we save a bit of memory if we only use one variable name :-)

**Comment by patrickvonplaten** [comment]
Awesome thanks @patil-suraj! 

Merging

**Comment by dleve123** [comment]
@patil-suraj Thanks for this! I'm working on a multi-task version of `LongformerForSequenceClassification`. For my context, why did you decide to implement pooling separately from the [pooling done in `LongformerModel`](https://github.com/huggingface/transformers/blob/main/src/transformers/models/longformer/modeling_longformer.py#L1377-L1383)? It seems like the key differences between the pooling done in `LongformerClassificationHead` vs. `LongformerPooler` are:

1. a dropout layer before the dense layer ([source](https://github.com/huggingface/transformers/blob/9ade58f0555430cec851e307c83c3a56c4a77d0b/src/transformers/models/longformer/modeling_longformer.py#L2004))
2. additional dropout and dense layers ([source](https://github.com/huggingface/transformers/blob/9ade58f0555430cec851e307c83c3a56c4a77d0b/src/transformers/models/longformer/modeling_longformer.py#L2007-L2008))

I see that this mimics the [`RobertaForSequenceClassification` implementation](https://github.com/huggingface/transformers/blob/main/src/transformers/models/roberta/modeling_roberta.py#L1449-L1468). Is the goal to avoid the pooler parameters learned during pre-training a `LongformerModel`? I see that this topic has been discussed in general (https://github.com/huggingface/transformers/issues/1328), but I am curious to learn more specifically for Longformer!
